feat(react-scheduler): add editing by dragging (#1870)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/drag-drop-provider.md

# DragDropProvider Plugin Reference

A plugin that enables users to edit appointments via drag-and-drop.

## Import

Use the following statement to import a plugin with embedded theme components:

```js
import { DragDropProvider } from '@devexpress/dx-react-scheduler-material-ui';
```

> If you want to use custom components, you can import the themeless plugin:

Reviewer: ```suggestion
To use custom components, import the themeless plugin:
```

Reviewer 2: This is a standard phrase that we use in every reference article

Reviewer: Ok, but you can make it shorter.

```js
import { DragDropProvider } from '@devexpress/dx-react-scheduler';
```

## User Reference

### Dependencies

- [EditingState](editing-state.md)
- [Appointments](week-view.md)
- [DayView](day-view.md) [Optional]
- [WeekView](week-view.md) [Optional]
- [MonthView](month-view.md) [Optional]
- [AllDayPanel](all-day-panel.md) [Optional]

### Properties

Name | Type | Default | Description
-----|------|---------|------------
allowDrag | (appointmentData: [AppointmentModel](./scheduler.md#appointmentmodel)) => boolean | () => true | A function that specifies draggable appointments.
draftAppointmentComponent | ComponentType&lt;[DragDropProvider.DraftAppointmentProps](#dragdropproviderdraftappointmentprops)&gt; | | A component that renders the appointment being dragged.

> sourceAppointmentComponent | ComponentType&lt;[DragDropProvider.SourceAppointmentProps](#dragdropprovidersourceappointmentprops)&gt; | | A component that renders the copy of the appointment being dragged in the former location.

Reviewer: ```suggestion
sourceAppointmentComponent | ComponentType&lt;[DragDropProvider.SourceAppointmentProps](#dragdropprovidersourceappointmentprops)&gt; | | A component that renders a copy of the appointment being dragged in the former location.
```

Reviewer: in the former location?

Reviewer 2: While you are dragging an appointment, a copy of it is displayed where the appointment has been. Once you drop the appointment, the copy disappears

Reviewer: ```suggestion
sourceAppointmentComponent | ComponentType&lt;[DragDropProvider.SourceAppointmentProps](#dragdropprovidersourceappointmentprops)&gt; | | A component that renders a copy of the appointment being dragged in its previous location.
```
